README.md

```diff
@@ -15,73 +15,90 @@ Due to the fact that hardly any information can be encoded in `Type`, we rely on
  * `GroupsCore.gens(G::Group)`: return a random-accessed collection of generators of `G`; if a group does not come with a generating set (or it may be prohibitively expensive to compute), one needs to alter `GroupsCore.hasgens(::Group) = false`.
  * `Base.rand(rng::Random.AbstractRNG, rs::Random.Sampler{GT}) where GT<:Group`: to enable asking for random group elements treating group as a collection, i.e. calling `rand(G, 2, 2)`.
 
 ## `GroupElement` methods
-##### Obligatory methods
- * `Base.parent(g::GroupElement)`: return the parent object of a given group element. Parent objects of the elements of the same group must be **identical** (i.e. ===).
+#### Obligatory methods
+ * `Base.parent(g::GroupElement)`: return the parent object of a given group element. Parent objects of the elements of the same group must be **identical** (i.e. `===`).
  * `GroupsCore.parent_type(::Type{<:GroupElement})`: given the type of an element return the type of its parent.
- * `GroupsCore.istrulyequal(g::GEl, h::GEl) where GEl<:GroupElement`: return the mathematical equality of group elements; by default the standard equality `==` calls this function.
- * `GroupsCore.hasorder(g::GroupElement)`: return `true` if `g` has finite order (without computing it)`.
+ * `GroupsCore.==(g::GEl, h::GEl) where GEl<:GroupElement`: return the mathematical equality of group elements;
+ * `GroupsCore.isfiniteorder(g::GroupElement)`: return `true` if `g` has finite order (possibly without computing it)`. If `isfiniteorder(g)` returns `false`, `order(g)` is required to throw `GroupsCore.InfiniteOrder` exception.
  * `Base.deepcopy_internal(g::GroupElement, ::IdDict)`: return a completely intependent copy of group element `g` **without copying its parent**; `isbits` subtypes of `GroupElement` need not to implement this method.
  * `Base.inv(g::GroupElement)`: return the group inverse of `g`.
  * `Base.:(*)(g::GEl, h::GEl) where GEl<:GroupElement`: the group binary operation on `g` and `h`.
 
 No further methods are strictly nessesary.
 
-## Implemented methods
+### Implemented methods
 Based on these methods only, the following functions in `GroupsCore` are implemented:
- * `Base.one(::GroupElement)`
- * `GroupsCore.order(g)`, `order(I::Type{<:Integer}, g)`
+ * `Base.one(g::GroupElement)` → `one(parent(g))`
+ * `GroupsCore.order(g)`, `order(I::Type{<:Integer}, g)` (naive implementation)
  * `Base.literal_pow(::typeof(^), g, Val{-1})` → `inv(g)`
- * `Base.:(/)(g, h)` → `g*inv(h)`
- * `Base.conj(g, h)`, `Base.:(^)(g, h)` → `inv(h)*g*h`
- * `Base.comm(g, h)` → `inv(h)*inv(g)*h*g` and its `Vararg` (`foldl`) version.
- * `Base.:(==)(g,h)` → `GroupsCore.istrulyequal(g, h)`
+ * `Base.:(/)(g, h)` → `g*h^-1`
+ * `Base.conj(g, h)`, `Base.:(^)(g, h)` → `h^-1*g*h`
+ * `Base.comm(g, h)` → `h^-1*g^-1*h*g` and its `Vararg` (`foldl`) version.
```

Comment:
The convention used in GAP is:
```suggestion
 * `Base.comm(g, h)` → `g^-1*h^-1*g*h` and its `Vararg` (`foldl`) version.
```
This is also what Wikipedia lists (not that this means too much) and also what I see in books on group theory most frequently; it is consistent with right actions.

People preferring left actions typically also prefer defining `[g,h] = g*h*g^{-1}*h^{-1}`.

I am not aware of any source using "your" definition, but of course they still may exist :-)

Reply:
OK, `g^-1*h^-1*g*h` is also what you write for `comm!` so perhaps we can agree on that ;-).

Reply:
yes, that was bug in the readme